src/Features/VisualBasic/Portable/CodeFixes/AddExplicitCast/VisualBasicAddExplicitCastCodeFixProvider.vb

```diff
@@ -0,0 +1,175 @@
+                Case BC30512, BC42016
+                    Dim argument = spanNode.GetAncestors(Of ArgumentSyntax).FirstOrDefault()
+                    If argument IsNot Nothing AndAlso argument.GetExpression.Equals(spanNode) Then
+                        ' spanNode is an argument expression
+                        Dim argumentList = argument.GetAncestorOrThis(Of ArgumentListSyntax)
+                        Dim invocationNode = argumentList.Parent
+
+                        mutablePotentialConversionTypes.AddRange(GetPotentialConversionTypes(semanticModel, root,
+                            argument, argumentList, invocationNode, cancellationToken))
+                    Else
+                        ' spanNode is a right expression in assignment operation
+                        Dim inferenceService = document.GetRequiredLanguageService(Of ITypeInferenceService)()
+                        Dim conversionType = inferenceService.InferType(semanticModel, spanNode, False, cancellationToken)
+                        mutablePotentialConversionTypes.Add(Tuple.Create(spanNode, conversionType))
+                    End If
+                Case BC30518, BC30519
+                    Dim invocationNode = spanNode.GetAncestors(Of ExpressionSyntax).FirstOrDefault(
+                        Function(node) Not node.ChildNodes.OfType(Of ArgumentListSyntax).IsEmpty())
+                    Dim argumentList = invocationNode.ChildNodes.OfType(Of ArgumentListSyntax).FirstOrDefault()
+
+                    ' spanArgument is null because the span is on the invocation identifier name according to BC30519
+                    mutablePotentialConversionTypes.AddRange(GetPotentialConversionTypes(semanticModel, root,
+                        spanArgument:=Nothing, argumentList, invocationNode, cancellationToken))
+            End Select
+
+            ' clear up duplicate types
+            potentialConversionTypes = FilterValidPotentialConversionTypes(semanticModel, mutablePotentialConversionTypes)
+            Return Not potentialConversionTypes.IsEmpty
+        End Function
+
+        Protected Overrides Function IsObjectCreationExpression(targetNode As ExpressionSyntax) As Boolean
+            Return targetNode.Kind() = SyntaxKind.ObjectCreationExpression
+        End Function
+
+        Protected Overrides Function ClassifyConversionExists(semanticModel As SemanticModel,
+                expression As ExpressionSyntax, type As ITypeSymbol) As Boolean
+            Return semanticModel.ClassifyConversion(expression, type).Exists
+        End Function
+
+        Protected Overrides Function IsConversionIdentity(semanticModel As SemanticModel,
+                expression As ExpressionSyntax, type As ITypeSymbol) As Boolean
+            Return semanticModel.ClassifyConversion(expression, type).IsIdentity
+        End Function
+
+        Protected Overrides Function GetArguments(argumentList As SyntaxNode) As SeparatedSyntaxList(Of SyntaxNode)
```

Comment:
can we make the parent generic on TArgumentListSyntax?

Reply:
There are 'ArgumentListSyntax' and 'AttributeArgumentListSyntax' in C#, I couldn't find a good way to do the generic type lol